awx/main/tasks/host_indirect.py

```diff
@@ -0,0 +1,200 @@
+import logging
+from typing import Tuple, Union
+
+import yaml
+
+import jq
+
+from django.utils.timezone import now, timedelta
+from django.conf import settings
+from django.db import transaction
+
+# Django flags
+from flags.state import flag_enabled
+
+from awx.main.dispatch.publish import task
+from awx.main.dispatch import get_task_queuename
+from awx.main.models.indirect_managed_node_audit import IndirectManagedNodeAudit
+from awx.main.models.event_query import EventQuery
+from awx.main.models import Job
+
+logger = logging.getLogger(__name__)
+
+
+class UnhashableFacts(RuntimeError):
+    pass
+
+
+def get_hashable_form(input_data: Union[dict, list, Tuple, int, float, str, bool]) -> Tuple[Union[Tuple, int, float, str, bool]]:
+    "Given a dictionary of JSON types, return something that can be hashed and is the same data"
```

Comment:
this is an interesting approach, and I'm wondering if the same could have been achieved with

```
json.dumps(input_data, sort_keys=True)
```

to get a hashable string on the input data

Reply:
nvm I think the lists would not be stable